internal/services/machinelearning/machine_learning_registry_resource.go

```diff
@@ -0,0 +1,555 @@
+	Identity                      []identity.ModelSystemAssignedUserAssigned `tfschema:"identity"`
+}
+
+type MachineLearningRegistryResource struct{}
+
+var _ sdk.ResourceWithUpdate = (*MachineLearningRegistryResource)(nil)
+
+func (m MachineLearningRegistryResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[a-zA-z0-9][a-zA-z0-9_-]{2,32}$`), "Registry name must be between 3 and 33 characters long. Its first character has to be alphanumeric, and the rest may contain hyphens and underscores. No whitespace is allowed."),
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"identity": commonschema.SystemOrUserAssignedIdentityOptionalForceNew(),
+
+		"location": commonschema.Location(),
+
+		"public_network_access_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			ForceNew: true,
+			Default:  false,
+		},
+
+		"region_details": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			MinItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"acr_details": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"user_created_acr_account_arm_resource_id": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ValidateFunc: validation.StringIsNotEmpty,
+								},
+							},
+						},
+					},
+
+					"location": commonschema.LocationWithoutForceNew(),
+
+					"storage_account_details": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"user_created_storage_account_arm_resource_id": {
+									Type:         pluginsdk.TypeString,
+									Optional:     true,
+									ValidateFunc: validation.StringIsNotEmpty,
+								},
+
+								"system_created_storage_account_type": {
+									Type:     pluginsdk.TypeString,
+									Optional: true,
+									ValidateFunc: validation.StringInSlice([]string{
+										"Standard_LRS",
+										"Standard_GRS",
+										"Standard_RAGRS",
+										"Standard_ZRS",
+										"Standard_GZRS",
+										"Standard_RAGZRS",
+										"Premium_LRS",
+										"Premium_ZRS",
+									}, false),
+								},
+
+								"system_created_storage_account_hns_enabled": {
+									Type:     pluginsdk.TypeBool,
+									Optional: true,
+									Default:  false,
+								},
+
+								"system_created_storage_account_arm_resource_id": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+
+								"system_created_storage_account_name": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
```

Comment:
Based on how the API is described/behaves we want to pull out and bring the properties that determine the primary region to the top of the schema.

The API also allows users to either specify a pre-existing container registry and storage account, or to omit them and let the API create these for the user. 

Whether we expose this functionality depends on how to API behaves and will determine whether `container_registry_id` and `storage_account_id` will be `Optional` or `Required`.

Can you please create a machine learning registry where the API creates the container registry and storage account for the user and check whether these are also deleted by the API when the machine learning registry is deleted. 

If the API reliably cleans up any artifacts and resources that it creates then we are fine to expose this functionality to end users and the properties can be `Optional`.

```suggestion
		"container_registry_id": {
			Type: pluginsdk.TypeString,
			// See review comment in GH
			// Required: 	true,
			// Optional:     true,
			ValidateFunc: registries.ValidateRegistryID,
		},

		"storage_account_id": {
			Type:          pluginsdk.TypeString,
			// See review comment in GH
			// Required: 	true,
			// Optional:     true,
			ValidateFunc:  validate.StorageAccountID,
			ConflictsWith: []string{"storage_account_type", "storage_account_hns_enabled"},
		},

		"storage_account_type": {
			Type:     pluginsdk.TypeString,
			Optional: true,
			ValidateFunc: validation.StringInSlice([]string{
				"Standard_LRS",
				"Standard_GRS",
				"Standard_RAGRS",
				"Standard_ZRS",
				"Standard_GZRS",
				"Standard_RAGZRS",
				"Premium_LRS",
				"Premium_ZRS",
			}, false),
			ConflictsWith: []string{"storage_account_id"},
		},

		"storage_account_hns_enabled": {
			Type:          pluginsdk.TypeBool,
			Optional:      true,
			Default:       false,
			ConflictsWith: []string{"storage_account_id"},
		},

		"replication_regions": {
			Type:     pluginsdk.TypeList,
			Required: true,
			// This might only be conditionally ForceNew? Please investigate the API to find out
			ForceNew: true,
			MinItems: 1,
			Elem: &pluginsdk.Resource{
				Schema: map[string]*pluginsdk.Schema{
					"container_registry_id": {
						Type:         pluginsdk.TypeString,
						// See review comment in GH
						// Required: 	true,
						// Optional:     true,
						ValidateFunc: registries.ValidateRegistryID,
					},

					"location": commonschema.Location(),

					"storage_account_id": {
						Type:         pluginsdk.TypeString,
						// See review comment in GH
						// Required: 	true,
						// Optional:     true,
						ValidateFunc: validate.StorageAccountID,
						ConflictsWith: []string{
							"replication_regions.0.system_created_storage_account_type",
							"replication_regions.0.storage_account_hns_enabled",
						},
					},

					"storage_account_type": {
						Type:     pluginsdk.TypeString,
						Optional: true,
						ValidateFunc: validation.StringInSlice([]string{
							"Standard_LRS",
							"Standard_GRS",
							"Standard_RAGRS",
							"Standard_ZRS",
							"Standard_GZRS",
							"Standard_RAGZRS",
							"Premium_LRS",
							"Premium_ZRS",
						}, false),
						ConflictsWith: []string{"replication_regions.0.storage_account_id"},
					},

					"storage_account_hns_enabled": {
						Type:          pluginsdk.TypeBool,
						Optional:      true,
						Default:       false,
						ConflictsWith: []string{"replication_regions.0.storage_account_id"},
					},
				},
			},
		},
```